### DISCUSSION TITLE
order by a `non selectable` field

### DISCUSSION THREAD

**Discussion by khlilturki97** [rebing/graphql-laravel#932] [Q&A]
hi,

I have a type whith a non selectable field 
```
                'active' => [
                    'type' => Type::boolean(),
                    'selectable' => false,
                    'resolve' => function ($model) {
                        return $model->start_date < Carbon::now() && $model->end_date > Carbon::now();
                    }
                ]
```

i want to sort the result by that field and use pagination.

without pagination, i can get the result as a collection and order it.

but with pagination i have to return a `Paginator`

Thanks

**Comment by mfn** [comment]
> i want to sort the result by that field and use pagination.

Then you have to do this _before_ you return your data in the resolve method.

Since you implemented this logic on a GraphQL fields resolver directly, you can't make use of that because the, assuming Eloquent models, don't know about it either.

But the collection of Eloquent models is where you would need to apply the sorting, hence the logic/data must be present there already. I would also expect you want a second sort criteria than just sort for `true`/`false`, so you should probably get back to your _Eloquent_ query for this.

Not sure I could describe it in way it's easy to understand; maybe I can help more with more details/more code examples.
